### ISSUE TITLE
bug: Story Logger - Failed to download

### ISSUE THREAD

**Issue by xnvmx** [rhunk/SnapEnhance#715]
### Description

I need to recover story. A few stories expired 3 hours ago. This stories was sent to a group of people. 

I can open all stories in the form of photos, but some of them cannot be saved with the message "Failed to download". 

One important story in the form of a video shows in the thumbnail, but when I want to open it - crashes and turns off SnapEnhance app (in log: Failed to get scripts folder) and when I try to save it also shows "Failed to download".

They are very important to me. 

Why can I see them in Story Logger but I can't download some of them?

Edit:
I noticed that the problem with opening video stories occurs with some stories that are in video form and have an overlay

### Reproduction steps

1. Go to `...`
2. Click on `....`
3. Scroll down to `....`
4. See error.

### Snapchat Version

12.57.0.55

### SnapEnhance Version

Newest Beta

### Agreement

- [X] This is not a bug regarding Snapchat+.
- [X] I have provided a detailed description of the issue.
- [X] I have attached a log if deemed neccessary.
- [X] This issue is not a duplicate.

**Comment by xnvmx** [comment]
@rhunk @authorisation

**Comment by authorisation** [comment]
Logcat? It's an experimental feature, stuff like this is expected to happen. You mentioned you can still see a preview of the image or whatever, that means it's cached but in bad quality (Better than nothing?).

**Comment by xnvmx** [comment]
I currently have a problem with one video story, it's the only one that won't open or download.

Well, I just installed the file manager and now when I click "open" it takes me to the cache folder, as you say. There is a separately saved video file with this story and its overlay with text. 

In the thumbnail in the Story Logger I can see it merged, but in the cache it is separate. How to save it merged?



BTW It's a pity that it's of lower quality, although it's not very noticeable, these are probably minimal differences, but you're right - it's better than nothing.
I love all of you who are working on this project, please tell me how I can support you.

**Comment by rhunk** [comment]
I've implemented a way to save stories from the cache even if the story has been deleted on their servers. Hope this help
https://github.com/rhunk/SnapEnhance/commit/7283063daf743a355a9775efb89cf3a7f2fede60

**Comment by xnvmx** [comment]
> I've implemented a way to save stories from the cache even if the story has been deleted on their servers. Hope this help [7283063](https://github.com/rhunk/SnapEnhance/commit/7283063daf743a355a9775efb89cf3a7f2fede60)

Unfortunately, it won't help, because I just opened the story logger and the stories are no longer there :(
But thank you for this improvement.

BTW I think I copied somewhere the one video story with which I had the biggest problem, but it is separate from the overlay and I don't know how I could merge it now.
And it would be nice if we could set the length of time that stories are kept in the story logger cache.
